src/swap/defi/xrpDex.ts

```diff
@@ -95,12 +96,25 @@ const fetchSwapQuoteInner = async (
   const rippleServers: string[] = RIPPLE_SERVERS_DEFAULT
   const volatilitySpread: number = VOLATILITY_SPREAD_DEFAULT
 
-  const fromIssuer = fromTokenId != null ? fromTokenId.split('-')[1] : undefined
-  const toIssuer = toTokenId != null ? toTokenId.split('-')[1] : undefined
-  const fromCurrency =
-    fromTokenId != null ? fromTokenId.split('-')[0] : fromCurrencyCode
-  const toCurrency =
-    toTokenId != null ? toTokenId.split('-')[0] : toCurrencyCode
+  let fromIssuer: string | undefined
+  let fromCurrency: string = fromCurrencyCode
```

Comment:
fromCurrencyCode is getting deprecated, we shouldn't be using it

Reply:
if tokenId == null then set the fromCurrency to the main asset of the wallet

 `wallet.currencyInfo.currencyCode`